### DISCUSSION TITLE
LoggerEngineDataSelector_Tests.it_returns_cached_tag_assignment_rules assertion error

### DISCUSSION THREAD

**Discussion by angelikagalang** [jongpie/NebulaLogger#799] [Q&A] [answered]
Hi! We have been using Nebula (unlocked) for quite a while now and did not encounter any issues except when we introduced a LogEntryTagRule metadata record.

This is both encountered when manually running the test and on our CI/CD build where a new scratch org is created for validation. As you can see, the expected `SObjectField__c` I think is the field definition ID while the actual is the field API name.

Tried using the latest package `v4.14.18` but encountered the same issue. Hope you could assist me, thanks in advance!

```
{
  "failedTests": [
    {
      "Id": "07M9K00000APLnLUAX",
      "QueueItemId": "7099K00000KqmTLQAZ",
      "StackTrace": "Class.LoggerEngineDataSelector_Tests.it_returns_cached_tag_assignment_rules: line 149, column 1",
      "Message": "System.AssertException: Assertion Failed: Expected: (LogEntryTagRule__mdt:{Id=m019K00000fqwLmQAI, SObjectField__c=01I9K000005xDcj.00N9K00000MLL8l, ComparisonType__c=CONTAINS, ComparisonValue__c=case review, Tags__c=case-review, SObjectType__c=01I9K000005xDcj}), Actual: (LogEntryTagRule__mdt:{Id=m019K00000fqwLmQAI, SObjectField__c=Message__c, ComparisonType__c=CONTAINS, ComparisonValue__c=case review, Tags__c=case-review, SObjectType__c=01I9K000005xDcj})",
      "AsyncApexJobId": "7079K00000axYAiQAM",
      "MethodName": "it_returns_cached_tag_assignment_rules",
      "Outcome": "Fail",
      "ApexClass": {
        "Id": "01p9K00000FAmV1QAL",
        "Name": "LoggerEngineDataSelector_Tests",
        "NamespacePrefix": null
      },
      "RunTime": 1102,
      "FullName": "LoggerEngineDataSelector_Tests.it_returns_cached_tag_assignment_rules"
    }
  ]
}
```

**Comment by jongpie** [comment, marked as answer]
@angelikagalang thanks for reporting this & apologies for the delay! I've logged issue #812 to track this - this also relates to some changes that I'm actively working on for issue #687, so I'm hoping to have it fixed soon (this week, maybe?).

**Reply by angelikagalang** [reply]
@jongpie Thanks for the reply! Much appreciated 🚀

**Reply by jongpie** [reply]
@angelikagalang this should now be fixed in [release `v4.15.3`](https://github.com/jongpie/NebulaLogger/releases/tag/v4.15.3) 🎉 Let me know if you see this error again (or run into any other problems).
